docs/infrastructure/microk8s/install.md

```diff
@@ -0,0 +1,274 @@
+2. Now that these plugins have been enabled, we should restart microk8s to ensure a proper bootstrap of the cluster: `microk8s.stop && microk8s.start && microk8s.status --wait-ready
+sleep 10`. Further information on this subject can be [found here](https://github.com/ubuntu/microk8s/issues/493#issuecomment-498167435).
+3. The plugins are now enabled and the cluster bootstrapped. However, we must still wait for some microk8s internals (dns, http, and ingress) plugins to be ready. Failing to do so can result in pods entering a `CrashLoopBackoff` state:
+   1. `microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns --timeout=300s`
+   2. `microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=hostpath-provisioner --timeout=300s`
+   3. `microk8s.kubectl wait -n default --for=condition=Ready pod -l app=default-http-backend --timeout=300s`
+   4. `microk8s.kubectl wait -n default --for=condition=Ready pod -l name=nginx-ingress-microk8s --timeout=300s`
+4. After runing these commands, we have ensured that dns, http, and nginx ingress are enabled and working properly inside the cluster.
+
+### Installing Codacy
+
+#### Helm and Tiller
+
+1. First we must install `helm` onto the cluster. With `helm`, we can easily install charts and manage the lifecycle of the installed artifacts. One example is the ability to rollback a failed install.
+2. Assuming you would want to install `helm` version 2.16.3, export the following environment variable: `export HELM_VERSION=2.16.3`
+3. Run this script that will retrieve the desired `helm` version and install it onto the machine's `/usr/local/bin`: `HELM_PKG=helm-${HELM_VERSION}-linux-amd64.tar.gz
+wget https://get.helm.sh/$HELM_PKG
+tar xvzf $HELM_PKG
+mv linux-amd64/tiller linux-amd64/helm /usr/local/bin
+rm -rvf $HELM_PKG linux-amd64/`.
+   1. Two executables will get installed: `helm` and `tiller`. The former is responsible for resolving the configuration of the chart to be installed, while issuing the correct install commands onto the cluster. The latter is responsible for receiving the install commands issued by `helm`, as well as managing the lifecycle of the components that have been installed. `helm` is the client facing side, while `tiller` is the server/cluster facing side.
+4. We must now deploy `tiller` onto the cluster.
+   1. First we must set a `serviceaccount` and a `clusterrolebinding` for `tiller` on the microk8s cluster: `microk8s.kubectl create serviceaccount --namespace kube-system tiller
+microk8s.kubectl create clusterrolebinding tiller-cluster-rule --clusterrole=cluster-admin --serviceaccount=kube-system:tiller`
+   2. Then we can install `tiller`, which will be installed with the same `HELM_VERSION` that you have specified earlier: `helm init --service-account tiller`
+   3. We wait for tiller to be ready: `microk8s.kubectl -n kube-system wait --for=condition=Ready pod -l name=tiller --timeout=300s`
```

Comment:
I might be wrong, but I think that for installing tiller we do not need dns etc.
We can probably install helm and tiller before enabling the "extra-services" in microk8s and than wait one time for all, WDYT?

Reply:
The only reason why it is in that order was just because the text seemed to flow better. First everything about microk8s, then everything helm related. I guess @pauloribeiro-codacy could help with this. :)